src/hovers/hoverHandler.ts

```diff
@@ -0,0 +1,43 @@
+import { ExtensionContext, languages, Position } from "vscode";
+import { getHover, Hover } from "../binary/requests/hovers";
+import { StatePayload } from "../consts";
+import setState from "../binary/requests/setState";
+import registerHoverCommands from "./hoverActionsHandler";
+import showTextDecoration, { isDecorationContains } from "./decorationState";
+
+let currentHover: Hover | null | undefined = null;
+
+languages.registerHoverProvider(
+  { pattern: "**" },
+  {
+    provideHover(_document, position) {
+      return handleHoverShown(position);
+    },
+  }
+);
+
+function handleHoverShown(position: Position) {
+  if (currentHover && isDecorationContains(position)) {
+    void setState({
+      [StatePayload.HOVER_SHOWN]: {
+        id: currentHover.id,
+        text: currentHover.message,
+        notification_type: currentHover.notification_type,
+        state: currentHover.state
+      },
+    });
+  }
+  return null;
```

Comment:
why?

Reply:
I think you can just delete this row

Reply:
i changed it a bit, but the `provideHover`  expects to  `vscode.ProviderResult` as a result
https://code.visualstudio.com/api/references/vscode-api#HoverProvider